### PR TITLE
Fix resuming when downloading in chunked mode

### PR DESCRIPTION
**Pull request by chrisww**
When `url_save` is passed a list of urls (`is_chunked = True`), and the download is aborted, then the next time that you-get resumes the download, it will append the video data from beginning to the `.download` file, because it does not send a correct `Range` header when resuming.

Several "exceeding 100%" issues might be related to this: #2703, #2768

Steps to reproduce:
1. `you-get "https://www.youtube.com/watch?v=Y8Tko2YC5hA"`
2. Press `Ctrl-C` before the download is finished
3. Re-start the `you-get` command
3. The downloaded size will eventually exceed the expected file size and some merging error will appear

<details>
  <summary>The transcript for the steps</summary>
  
```
~/test > you-get "https://www.youtube.com/watch?v=Y8Tko2YC5hA"
site:                YouTube
title:               What is Python and Why You Must Learn It in [2019]
stream:
    - itag:          248
      container:     webm
      quality:       1920x1080 (1080p)
      size:          17.3 MiB (18168269 bytes)
    # download-with: you-get --itag=248 [URL]

Downloading What is Python and Why You Must Learn It in (2019).webm ...
53.4% (  9.2/ 17.3MB) ├███████████████████████████████████████████████████████████████████───────────────────────────────────────────────────────────┤[1/2]   13 MB/s^C¶
~/test (1) > you-get "https://www.youtube.com/watch?v=Y8Tko2YC5hA"
site:                YouTube
title:               What is Python and Why You Must Learn It in [2019]
stream:
    - itag:          248
      container:     webm
      quality:       1920x1080 (1080p)
      size:          17.3 MiB (18168269 bytes)
    # download-with: you-get --itag=248 [URL]

Downloading What is Python and Why You Must Learn It in (2019).webm ...
 100% ( 23.4/ 17.3MB) ├██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████┤[2/2]   11 MB/s
Merging video parts... Merging without re-encode failed.
Try again re-encoding audio... Merged into What is Python and Why You Must Learn It in (2019).webm

Saving What is Python and Why You Must Learn It in (2019).en.srt ... Done.
```
</details>

Basically, the idea of this pull request is to maintain the starting and ending offset for each chunk. And when resuming, it skips downloaded chunks by comparing the file size and ending offsets, and computes the correct `Range` for the first unfinished chunk.

### CODE REVIEW

**Comment by soimort** [comment]
Thanks a lot for the fix!
